src/components/PrivateRoute.jsx

```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+import { Redirect, Route } from 'react-router-dom'
+import { useUser } from '../context/UserContext';
+
+export default function PrivateRoute({ children, ...rest }) {
+    const user = useUser();
+    return (
+        <div>
+               <Route {...rest} render={( privateRouteProps ) => 
```

Comment:
This actually doesn't need to be wrapped in a `div`. The `Route` component is self-contained so this is just an extra wrapper.